modules/account-lib/src/coin/hbar/tokenTransferBuilder.ts

```diff
@@ -0,0 +1,115 @@
+      if (!(tokenName in tokenTransferData)) {
+        tokenTransferData[tokenName] = [
+          {
+            accountID: this.buildAccountData(this._source.address),
+            amount: Long.fromInt(0),
+          },
+        ];
+      }
+      tokenTransferData[tokenName].push(
+        { accountID: this.buildAccountData(recipient.address), amount: Long.fromString(recipient.amount) }, // recipient
+      );
+    });
+
+    Object.entries(tokenTransferData).forEach(([tokenName, tokenData]) => {
+      tokenData[0].amount = Long.fromString(tokenTransferAmount[tokenName].toString()).negate(); // update sender send amount
+      const tokenId = getHbarTokenFromTokenName(tokenName)!.tokenId!;
+      tokenTransfers.push({ token: this.buildTokenId(tokenId), transfers: tokenData });
+    });
+
+    return tokenTransfers;
+  }
+
+  private buildTokenId(tokenId: string): proto.TokenID {
+    const tokenData = TokenId.fromString(tokenId);
+    return new proto.TokenID({
+      tokenNum: tokenData.num,
+      realmNum: tokenData.realm,
+      shardNum: tokenData.shard,
+    });
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.tokenTransfers && transferData.tokenTransfers.length > 0) {
+      this.initTokenTransfers(transferData.tokenTransfers);
+    }
+  }
+
+  /**
+   * Initialize the transfer specific data, getting the recipient account
+   * represented by the element with a positive amount on the transfer element.
+   * The negative amount represents the source account so it's ignored.
+   *
+   * @param {proto.IAccountAmount[]} transfers array of objects which contains accountID and transferred amount
+   */
+  protected initTokenTransfers(tokenTransfers: proto.ITokenTransferList[]): void {
+    const tokenTransfer = tokenTransfers[0];
+    if (!tokenTransfer.token) {
+      throw new BuildTransactionError('Invalid transaction: missing token id');
+    }
+    if (!tokenTransfer.transfers) {
+      throw new BuildTransactionError('Invalid transaction: missing transfer data');
+    }
+    tokenTransfer.transfers.forEach((transferData) => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        const token = getHbarTokenFromId(stringifyTokenId(tokenTransfer.token!), this._coinConfig.network);
+        if (!token) {
+          throw new BuildTransactionError('Invalid transaction: invalid token id');
+        }
+        this.send({
+          address: stringifyAccountId(transferData.accountID!),
+          amount: amount.toString(),
+          tokenName: token.name,
+        });
+      }
+    });
+  }
+
+  // region Transfer fields
+  /** @inheritdoc */
+  send(recipient: Recipient): this {
+    if (!recipient.tokenName) {
+      throw new InvalidParameterValueError('Invalid missing token name');
```

Comment:
The `InvalidParameterValueError` throws should be updated to `BuildTransactionError` in this method.